Updates tests.

BE-Webdesign opened the pull request:
Closes #938.  The other PRs have been updated for better conventions.
Here is an update to the tests already existing in the repo.  Updates
FormToggle, Dashicon, and Button.

**Testing Instructions**
Verify tests still pass.

BE-Webdesign committed:
Updates tests.

Closes #938.  The other PRs have been updated for better conventions.
Here is an update to the tests already existing in the repo.  Updates
FormToggle, Dashicon, and Button.

**Testing Instructions**
Verify tests still pass.

nylen commented on components/form-toggle/test/index.js:
```diff
@@ -22,39 +22,39 @@ describe( 'FormToggle', () => {
 		} );
 
-		it( 'with additonal className', () => {
+		it( 'shouuld render with an additional className', () => {
```
Typo: `shouuld` &rarr: `should`

nylen commented on components/button/test/index.js:
```diff
@@ -24,71 +24,67 @@ describe( 'Button', () => {
+		it( 'should render and additional WordPress prop of value awesome', () => {
+			const button = <Button WordPress="awesome" />;
+
+			expect( button.props.WordPress ).to.equal( 'awesome' );
```
Should this be `button.prop( 'WordPress' )`?

BE-Webdesign replied:
I don't think so.  I admit it is probably confusing.  This is just pure React, I don't believe Enzyme is touching this in anyway. I believe it would need to be wrapped in `shallow` for the API to be available.  I should probably just wrap it in shallow and do what you said.

nylen commented on components/form-toggle/test/index.js:
```diff
@@ -22,39 +22,39 @@ describe( 'FormToggle', () => {
 		} );
 
-		it( 'with onChange handler', () => {
+		it( 'should render a checkbox with a noop on change and if an onChange function is provided override the noop', () => {
```
I think this should be two separate tests:

```
should render a checkbox with a noop onChange
should render a checkbox with a user-provided onChange
```

BE-Webdesign replied:
Sounds good.

nylen reviewed (approved):
Feel free to merge after addressing inline notes. :+1:

aduth commented:
I can probably be partly to blame for perpetuating the "should" convention for starting test case descriptions. I don't feel so strongly about it these days. I think a more reasonable convention, especially with the method name, is to simply form it as a complete sentence, which _could_ be `it( 'should render ...`, but could equally be `it( 'renders ...`.

Two cents. Dunno how others feel.

nylen commented:
My preference is simply that the test names form a complete and readable sentence as well.  I usually write `should` just out of habit and adhering to established <sup>citation needed</sup> BDD conventions.

BE-Webdesign committed:
Add revisions.

Adding revisions to the tests.  Cleaning up Enzyme API usage, and
breaking larger tests into more test cases.  Spelling corrections added.

aduth commented:
>I usually write `should` just out of habit and adhering to established <sup>citation needed</sup> BDD conventions.

I blame Mocha's docs 😛  https://mochajs.org/#getting-started
